### PR TITLE
Fix race condition in TestControlMapPauseState

### PR DESCRIPTION
**Pull request by alfrunes**

### CODE REVIEW

File: app/state_test.go [hunk 1]
```diff
@@ -5511,7 +5511,7 @@ func TestControlMapPauseState(t *testing.T) {
 	}
 	c := &stateTestController{
 		controlMap:      pool,
-		updatePollIntvl: 30,
+		updatePollIntvl: 30 * time.Second,
```
Comment (kacf) [inline app/state_test.go:5514]: Lol, that's one impatient test!

File: app/state_test.go [hunk 2]
```diff
@@ -1148,7 +1148,7 @@ type stateTransitionsWithUpdateModulesTestCase struct {
 
 var stateTransitionsWithUpdateModulesTestCases []stateTransitionsWithUpdateModulesTestCase = []stateTransitionsWithUpdateModulesTestCase{
 
-	stateTransitionsWithUpdateModulesTestCase{
+	{
```
Comment (kacf) [inline app/state_test.go:1151]: Yes!

But just FYI, IIRC I originally coded it like this, but had to add the explicit types due to old Golang versions in Yocto. That was a while back though, if dunfell is ok with this, then we are good to go.

File: app/state.go [hunk 3]
```diff
@@ -34,6 +35,49 @@ const (
 		"by currently installed artifact (%v != %v)."
 )
 
+type timerPool sync.Pool
```
Comment (kacf) [inline app/state.go:38]: Nothing against this change, but the performance impact was probably more important before the nanosecond fix! 😀

**Review by kacf — APPROVED** [review]

File: app/state.go (hunk 3 above)
Comment (alfrunes) [reply app/state.go:38]: Yeah, it was just a random thing I came across as I traced the code for the unit test. I was contemplating leaving it out of the PR, but decided to leave it in anyway.

File: app/state_test.go (hunk 2 above)
Comment (alfrunes) [reply app/state_test.go:1151]: I usually run `go fmt` with the `-s` option. The only time you have to use explicit struct declarations are for slices of interfaces as far as I know.

**Comment by alfrunes** [comment]
@kacf The client acceptance tests passed and the integration test failures are known spurious and not related specifically to the client alone. Do you think we can merge this one?

**Comment by kacf** [comment]
> @kacf The client acceptance tests passed and the integration test failures are known spurious and not related specifically to the client alone. Do you think we can merge this one?

Yes! 🙂

**Pull request merged**